src/crypto/mod.rs

```diff
@@ -0,0 +1,52 @@
+//! (Signing) cryptography abstraction
+
+use openssl::{hash::MessageDigest, nid::Nid};
+
+use crate::{error::CoseError, sign::SignatureAlgorithm};
+
+#[cfg(feature = "key_openssl_pkey")]
+mod openssl_pkey;
+
+/// A public key that can verify an existing signature
+pub trait SigningPublicKey {
+    /// This returns the signature algorithm and message digest to be used for this
+    /// public key.
+    fn get_parameters(&self) -> Result<(SignatureAlgorithm, MessageDigest), CoseError>;
+
+    /// Given a vector of data and a signature, returns a boolean whether the signature
+    /// was valid.
+    fn verify(&self, data: &[u8], signature: &[u8]) -> Result<bool, CoseError>;
```

Comment:
The `data` in this case seems to actually be the hash of the data, and not the data itself, based on the implementation. I propose that the field should either be called `hash` or `digest` or internalize the hashing to the function itself. This seems to be true for `sign` operation as well.

Reply:
True, will update.